model/common/src/icon4py/model/common/grid/grid_manager.py

```diff
@@ -82,22 +51,23 @@ class GridManager:
 
     def __init__(
         self,
-        transformation: IndexTransformation,
         grid_file: pathlib.Path | str,
-        config: v_grid.VerticalGridConfig,  # TODO(halungge): remove to separate vertical and horizontal grid
+        num_levels: int,
```

Comment:
We should get rid of the num_levels here, it is now only here because we have it in the `BaseGrid` which historically comes from the fact that we had it in the `SimpleMesh`